packages/cozy-harvest-lib/src/connections/accounts.js

```diff
@@ -0,0 +1,112 @@
+const ACCOUNTS_DOCTYPE = 'io.cozy.accounts'
+const PERMISSIONS_DOCTYPE = 'io.cozy.permissions'
+
+/**
+ * Performs call to create an io.cozy.accounts document
+ * @param  {Object}  client     CozyClient
+ * @param  {Object}  attributes Account attributes
+ * @return {Object}             The io.cozy.accounts document just created
+ */
+const createAccount = async (client, attributes) => {
+  const { data } = await client.create(ACCOUNTS_DOCTYPE, attributes)
+  return data
+}
+
+/**
+ * Performs a query to find the given account
+ * @param  {Object}  client CozyClient
+ * @param  {string}  id     io.cozy.accounts document's id
+ * @return {Object}         Retrieved account
+ */
+const findAccount = async (client, id) => {
+  const { data } = await client.query({ ...client.where(ACCOUNTS_DOCTYPE), id })
+  return data[0]
+}
+
+/**
+ * Creates an account for an konnector needing an aggregator account.
+ * Creates the aggregator account as well if it does not exist.
+ * @param  {Object}  client     CozyClient
+ * @param  {Object}  konnector  io.cozy.konnectors document
+ * @param  {Object}  attributes Account attibutes
+ * @return {Object}             The created io.cozy.accounts document
+ */
+const createChildAccount = async (client, konnector, attributes) => {
+  const { aggregator } = konnector
+
+  if (!aggregator || !aggregator.accountId)
+    throw new Error('Konnector does not provide aggregator account id')
+
+  const parentAccountId = aggregator.accountId
+
+  let parentAccount
+  try {
+    parentAccount = await findAccount(client, parentAccountId)
+  } catch (error) {
+    throw new Error(
+      `An error occurred when finding parent account ${parentAccountId} (${
+        error.message
+      })`
+    )
+  }
+
+  if (!parentAccount) {
+    try {
+      parentAccount = await createAccount(client, { _id: parentAccountId })
+    } catch (error) {
+      throw new Error(
+        `Cannot create parent account ${parentAccountId} (${error.message})`
+      )
+    }
+  }
+
+  try {
+    await client.collection(PERMISSIONS_DOCTYPE).add(konnector, {
+      aggregatorAccount: {
+        type: ACCOUNTS_DOCTYPE,
+        verbs: ['GET', 'PUT'],
+        values: [`${ACCOUNTS_DOCTYPE}.${parentAccountId}`]
+      }
+    })
+  } catch (error) {
+    throw new Error(
+      `Cannot set permission for account ${parentAccountId} (${error.message})`
+    )
+  }
+
+  return await createAccount(client, {
+    ...attributes,
+    relationships: {
+      parent: {
+        data: {
+          _id: parentAccountId,
+          _type: ACCOUNTS_DOCTYPE
+        }
+      }
+    }
```

Comment:
I don"t like to create manually the relationship so I tend to use cozy-client to create it (addById or something similar). Don't know if it's a best practise or not.

Reply:
I tried to use `addById` but it did not work immediatly. Next time I will investigate further.